app/src/main/java/org/simple/clinic/summary/PatientSummaryScreen.kt

```diff
@@ -172,6 +175,20 @@ class PatientSummaryScreen(
     super.onDetachedFromWindow()
   }
 
+  private fun teleconsultPhoneNumberSelected(): Observable<PatientSummaryEvent> {
+    return screenRouter
+        .streamScreenResults()
+        .ofType<ActivityResult>()
+        .filter { it.requestCode == CONTACT_DOCTOR_SHEET && it.succeeded() && it.data != null }
+        .map { it.data!! }
```

Comment:
We have this exact same codeblock everywhere where the only thing that changes is the type of the result data and the result code. Maybe we should think of moving this to an extension function.

```kotlin
inline fun <reified T> Observable<ActivityResult>.extractSuccessful(
    requestCode: Int,
    crossinline resultMapper: (Intent) -> T
): Observable<T> {
  return filter { it.requestCode == requestCode && it.succeeded() && it.data != null }
      .map { resultMapper(it.data!!) }
}
```

The usage then becomes a lot easier to read and developers can focus on the business logic instead of this boilerplate.

```kotlin
screenRouter
        .streamScreenResults()
        .ofType<ActivityResult>()
        .extractSuccessful(CONTACT_DOCTOR_SHEET, ContactDoctorSheet.Companion::readPhoneNumberExtra)
        .map { teleconsultPhoneNumberString ->
          val teleconsultPhoneNumber = TeleconsultPhoneNumber(teleconsultPhoneNumberString)

          ContactDoctorPhoneNumberSelected(teleconsultPhoneNumber)
        }
```

Reply:
Should this be part of this PR?

Reply:
Not necessary.

Reply:
Can we create a ticket for this?

Reply:
Done: https://www.pivotaltracker.com/story/show/173282881